source/2_platform/rex_windows/src/gui_application.cpp

```diff
@@ -15,12 +15,46 @@
 
 #include <Windows.h>
 
-// NOLINTBEGIN(modernize-use-nullptr,-warnings-as-errors)
+// NOLINTBEGIN(cppcoreguidelines-pro-type-union-access)
+// NOLINTBEGIN(modernize-use-nullptr)
 
 namespace rex
 {
   namespace win32
   {
+    class GuiApplicationState
+    {
+    public:
+      enum class Type
+      {
+        Resizing  = BIT(0),
+        Minimized = BIT(1),
+        Maximized = BIT(2)
+      };
+
+      GuiApplicationState()
+          : m_state(0u)
+      {
+      }
+
+      void add_state(GuiApplicationState::Type state)
```

Comment:
these functions look like they'd be more benefitial as library functions so they can be used for any enum